ci_scripts/build_macos.sh

```diff
@@ -76,8 +76,9 @@ cmake_args=("-G" "Xcode" \
   "-D" "KTX_FEATURE_TESTS=$FEATURE_TESTS" \
   "-D" "KTX_FEATURE_TOOLS=$FEATURE_TOOLS" \
   "-D" "KTX_FEATURE_TOOLS_CTS=$FEATURE_TOOLS_CTS" \
-  "-D" "KTX_LOADTEST_APPS_USE_LOCAL_DEPENDENCIES=$LOADTESTS_USE_LOCAL_DEPENDENCIES"
-  "-D" "KTX_WERROR=$WERROR"
+  "-D" "KTX_LOADTEST_APPS_USE_LOCAL_DEPENDENCIES=$LOADTESTS_USE_LOCAL_DEPENDENCIES" \
+  "-D" "KTX_WERROR=$WERROR" \
+  "-D" "CMAKE_EXPORT_COMPILE_COMMANDS=ON" \
```

Comment:
What is `CMAKE_EXPORT_COMPILE_COMMANDS` for?

Sorry about the missing backslashes. I'm surprised our CI passed.

Reply:
`CMAKE_EXPORT_COMPILE_COMMANDS` is used to generate `compile_commands.json` which is used by IDE's and tools like `clangd` to understand how to provide completions etc in `emacs` and `vim` if used with the right plugins. It doesn't have to be commited but it also doesn't hurt, there is no performance impact of this. Its only used in `configure` time. In this instance I had to change the generator from `Xcode` to `Ninja` for this to work. So it won't work as it us anyways. I will remove it in the next commits.

Reply:
Removed it for now.